lib/rucio/common/schema/generic.py

```diff
@@ -54,7 +55,7 @@
                      "T0 Export", "T0 Tape", "Upload/Download (Job)",
                      "Upload/Download (User)", "User Subscriptions", "Data Challenge"]}
 
-SCOPE_LENGTH = 25
+SCOPE_LENGTH = 25 * 40
```

Comment:
Why?
This would certainly be a problem to change, since the models.py uses this value the instantiate the length of the database columns. So you increased this column size by a factor of 40 by this :-)

Reply:
I definitely went way overboard with the new size. But we need to bump this number a bit. Here's an example that fails (117 characters).
```
[Mon Feb 21 12:57:15.876101 2022] [wsgi:error] [pid 32:tid 140136349914880] [remote 10.100.143.192:58938] Failed validating 'pattern' in schema['properties']['scope']:
[Mon Feb 21 12:57:15.876103 2022] [wsgi:error] [pid 32:tid 140136349914880] [remote 10.100.143.192:58938]     {'description': 'Scope name',
[Mon Feb 21 12:57:15.876106 2022] [wsgi:error] [pid 32:tid 140136349914880] [remote 10.100.143.192:58938]      'pattern': '^[a-zA-Z_\\\\-.0-9]{1,25}$',
[Mon Feb 21 12:57:15.876108 2022] [wsgi:error] [pid 32:tid 140136349914880] [remote 10.100.143.192:58938]      'type': 'string'}
[Mon Feb 21 12:57:15.876111 2022] [wsgi:error] [pid 32:tid 140136349914880] [remote 10.100.143.192:58938]
[Mon Feb 21 12:57:15.876113 2022] [wsgi:error] [pid 32:tid 140136349914880] [remote 10.100.143.192:58938] On instance['scope']:
[Mon Feb 21 12:57:15.876117 2022] [wsgi:error] [pid 32:tid 140136349914880] [remote 10.100.143.192:58938]     'valid1.361107.PowhegPythia8EvtGen_AZNLOCTEQ6L1_Zmumu.merge.NTUP_PHYSVAL.e5112_s3214_d1731_r13252_p4889_tid27512986_00'
```

Shall I change `SCOPE_LENGTH` to 120?

Reply:
This is not a valid scope, I think we need to separate the schema scope_length from the traces scope length.
In tis example, this is just wrong and the scope for that should be `valid1`, but that's not something we can enforce now.

Reply:
This is not a valid scope, I think we need to separate the schema scope_length from the traces scope length.
In tis example, this is just wrong and the scope for that should be `valid1`, but that's not something we can enforce now.

Reply:
Aye, I can revert this tomorrow i.e. set SCOPE_LENGTH back to 25 and check if the logs blow up again.